march_hardware/include/march_hardware/MarchRobot.h

```diff
@@ -58,6 +60,13 @@ class MarchRobot
 
   Joint& getJoint(::std::string jointName);
 
+  Joint& getJoint(size_t index);
+
+  size_t size() const;
+
+  iterator begin();
+  iterator end();
```

Comment:
I would go for a more descriptive name if not to much trouble, begin and end could also refer to something like the initiating of the robot.
```suggestion
  size_t JointListSize() const;

  iterator JointListBegin();
  iterator JointListEnd();
```

Reply:
the `begin` and `end` functions have to have that name otherwise the for each loop won't work. I have named the other `size` to be consistent with `std::vector`.